graylog2-server/src/main/java/org/graylog2/rest/resources/system/contentpacks/ContentPackResource.java

```diff
@@ -115,7 +117,8 @@ public ContentPackList listLatestContentPacks() {
         checkPermission(RestPermissions.CONTENT_PACK_READ);
 
         Set<ContentPack> contentPacks = contentPackPersistenceService.loadAllLatest();
```

Comment:
This endpoint is loading the latest revisions only. The `getInstallationMetaData()` method is loading all content packs though and will include the metadata for all content pack revisions.

How about passing the set of content packs into the `getInstallationMetaData()` method and only include the metadata for those content packs? That would avoid loading and including the metadata for all content packs and revisions all the time.

Reply:
We want to display if a content pack is installed regardless the revision. The `latest` endpoint will show all content packs but only the latest revision. So passing the ContentPacks to filter would not do any good.

Reply:
We are currently loading all content packs from the backend and do the pagination in the frontend. Ususally we use [PaginatedDbService](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/database/PaginatedDbService.java) to do the pagination and filtering in the backend to avoid loading all entries from the database all the time.

Once we use that, we don't have to load all installations but only the ones we need for the content packs on the active page.